api/types.go

```diff
@@ -12,33 +15,54 @@ type ProxyConfig struct {
 }
 
 type OIDCConfig struct {
-	IssuerUrl    *string
-	ClientId     *string
-	ClientSecret *string
-	Audience     *string
-	Enabled      *bool
+	Enabled    *bool
+	Filename   *string
+	Attributes OIDCAttributes
+}
+
+type OIDCAttributes struct {
+	IssuerURL    *string `yaml:"issuer_url"`
```

Comment:
Is there some particular reason the elements are pointers?

Reply:
The flags packages requires the target to be a pointer, but ff these are the attributes read from file, then they don't need to be pointers.

Reply:
The only reason I'd keep the pointer is if we want to differentiate between not being set vs being set as empty

Reply:
@pb82 @valeriiashapoval could we reevaluate this to see what is the most appropriate way? thanks

Reply:
Yes, that makes sense. There are no pointer required in that struct. All fields are strings, so they can already be empty (e.g. the audience).

@valeriiashapoval I think all you need to do is change them from pointers to non-pointer strings.